src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java

```diff
@@ -0,0 +1,92 @@
+import org.mockito.junit.MockitoJUnitRunner;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashApiClient;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashPullRequestResponseValue;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashPullRequestResponseValueRepository;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashPullRequestResponseValueRepositoryRepository;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StashRepositoryTest {
+
+  @Mock private StashBuildTrigger trigger;
+  @Mock private AbstractProject<?, ?> project;
+  @Mock private StashApiClient stashApiClient;
+  @Mock private StashPullRequestResponseValue pullRequest;
+  @Mock private StashPullRequestResponseValueRepository repository;
+  @Mock private StashPullRequestResponseValueRepositoryRepository repoRepo;
+
+  @Test
+  public void getTargetPullRequestsReturnsEmptyListForNoPullRequests() {
+    StashRepository stashRepository = new StashRepository(project, trigger, stashApiClient);
+
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.emptyList());
+
+    Collection<StashPullRequestResponseValue> buildablePullRequests =
+        stashRepository.getTargetPullRequests();
+    assertThat(buildablePullRequests, empty());
+
+    verify(stashApiClient, times(1)).getPullRequests();
+  }
+
+  @Test
+  public void getTargetPullRequestsAcceptsOpenPullRequests() {
+    StashRepository stashRepository = new StashRepository(project, trigger, stashApiClient);
+
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(trigger.getCiSkipPhrases()).thenReturn("NO TEST");
+    when(pullRequest.getFromRef()).thenReturn(repository);
+    when(pullRequest.getToRef()).thenReturn(repository);
+    when(repository.getRepository()).thenReturn(repoRepo);
+
+    Collection<StashPullRequestResponseValue> buildablePullRequests =
+        stashRepository.getTargetPullRequests();
+    assertThat(buildablePullRequests, contains(pullRequest));
+
+    verify(stashApiClient, times(1)).getPullRequests();
+    verify(pullRequest, times(1)).getTitle();
+  }
+
+  @Test
+  public void getTargetPullRequestsSkipsClosedPullRequests() {
+    StashRepository stashRepository = new StashRepository(project, trigger, stashApiClient);
+
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
+    when(pullRequest.getState()).thenReturn("CLOSED");
+
+    Collection<StashPullRequestResponseValue> buildablePullRequests =
+        stashRepository.getTargetPullRequests();
+    assertThat(buildablePullRequests, empty());
+
+    verify(stashApiClient, times(1)).getPullRequests();
+    verify(pullRequest, times(0)).getTitle();
+  }
+
+  @Test
+  public void getTargetPullRequestsSkipsNullStatePullRequests() {
+    StashRepository stashRepository = new StashRepository(project, trigger, stashApiClient);
```

Comment:
you seem to use the same setup, why not make it an instance field and use a @Before method?

Reply:
Done. For whatever reason, I started seeing some strange exceptions about missing Jenkins config file, so I added a `JenkinsRule()` call, which resolved them.